File: UI/angular11/src/app/dtos/order/order-read-dto.ts
import { CityReadDto } from "../city/city-read-dto"
import { CommentReadDto } from "../comment/comment-read-dto"
import { ReportReadDto } from "../report/report-read-dto"
import { UserReadDto } from "../user/user-read-dto"

export class OrderReadDto {
    id: number
    creationDate: Date
    prePaid: number
    postPaid: number
    user: UserReadDto
    demands: string
    isCompleted: boolean
    isStarted: boolean
    isRoot: boolean
    city: CityReadDto

    subOrders: OrderReadDto[]
    comments: CommentReadDto[]
    reports: ReportReadDto[]
}

File: ConstructionQualityControl.Domain/Dtos/Order/OrderReadDto.cs
using System;
using System.Collections.Generic;

namespace ConstructionQualityControl.Domain.Dtos
{
    public class OrderReadDto
    {
        public int Id { get; set; }
        public DateTime CreationDate { get; set; }
        public decimal PrePaid { get; set; }
        public decimal PostPaid { get; set; }
        public UserReadDto User { get; set; }
        public string Demands { get; set; }
        public bool IsCompleted { get; set; }
        public bool IsStarted { get; set; }
        public bool IsRoot { get; set; }
        public CityReadDto City { get; set; }

        public List<OrderReadDto> SubOrders { get; set; }
        public List<CommentReadDto> Comments { get; set; }
        public List<ReportReadDto> Reports { get; set; }
    }
}
